cos-gpu-installer-docker/entrypoint.sh

```diff
@@ -99,23 +99,42 @@ load_etc_os_release() {
 configure_kernel_module_locking() {
   info "Checking if third party kernel modules can be installed"
   local -r kernel_cmdline="$(cat /proc/cmdline)"
-  # Assume that kernel commandline will never contain "lsm.module_locking=1",
-  # which is the default value when unspecified.
-  if echo "${kernel_cmdline}" | grep -q -v "lsm.module_locking=0"; then
-    local -r esp_partition="/dev/sda12"
-    local -r mount_path="/tmp/esp"
-    local -r grub_cfg="efi/boot/grub.cfg"
-
-    mkdir -p "${mount_path}"
-    mount "${esp_partition}" "${mount_path}"
-
-    pushd "${mount_path}"
+  local -r esp_partition="/dev/sda12"
+  local -r mount_path="/tmp/esp"
+  local -r grub_cfg="efi/boot/grub.cfg"
+  local need_reboot="false"
+
+  mkdir -p "${mount_path}"
+  mount "${esp_partition}" "${mount_path}"
+  pushd "${mount_path}"
+
+  # COS m77+ uses module signature to verify kernel modules.
+  if echo "${kernel_cmdline}" | grep -q "module.sig_enforce"; then
+    if echo "${kernel_cmdline}" | grep -q "module.sig_enforce=1"; then
+      cp "${grub_cfg}" "${grub_cfg}.orig"
+      sed 's/module.sig_enforce=1/module.sig_enforce=0/g' \
+        -i "efi/boot/grub.cfg"
+      need_reboot="true"
+    fi
+  # Before m77, COS uses loadpin to verify kernel modules.
```

Comment:
This comment is a bit misleading, because the `else` case covers versions before M77. Maybe reorder the cases?

Reply:
I reworded the comment to make it COS version transparent.